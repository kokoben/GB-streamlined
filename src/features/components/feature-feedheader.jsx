import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchFeatureSearchVideos, setFeatureVideo } from '../actions/index';
import FeatureSearchResultsSelector from '../selectors/current-feature-search-results';

export const FeatureFeedHeader = props => (
  <FeedHeader
    category="Feature"
    placeholder="Search for Feature videos..."
    fetchVideos={props.fetchFeatureSearchVideos}
    results={props.featureSearchResults}
    currentResults={props.currentFeatureSearchResults}
    setVideo={props.setFeatureVideo}
  />
);

/* eslint-disable react/forbid-prop-types */
FeatureFeedHeader.propTypes = {
  fetchFeatureSearchVideos: PropTypes.func.isRequired,
  featureSearchResults: PropTypes.array.isRequired,
  currentFeatureSearchResults: PropTypes.array.isRequired,
  setFeatureVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  featureSearchResults: state.features.featureSearchResults,
  currentFeatureSearchResults: FeatureSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchFeatureSearchVideos,
    setFeatureVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(FeatureFeedHeader);
