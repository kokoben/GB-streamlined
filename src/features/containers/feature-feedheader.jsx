import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchFeatureSearchVideos, setFeatureVideo } from '../actions/index';
import FeatureSearchResultsSelector from '../selectors/current-feature-search-results';

class FeatureFeedHeader extends Component {
  render() {
    return (
      <FeedHeader
        category="Feature"
        placeholder="Search for Feature videos..."
        fetchVideos={this.props.fetchFeatureSearchVideos}
        currentResults={this.props.currentFeatureSearchResults}
        setVideo={this.props.setFeatureVideo}
      />
    );
  }
}

/* eslint-disable react/forbid-prop-types */
FeatureFeedHeader.propTypes = {
  fetchFeatureSearchVideos: PropTypes.func.isRequired,
  currentFeatureSearchResults: PropTypes.array.isRequired,
  setFeatureVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  currentFeatureSearchResults: FeatureSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchFeatureSearchVideos,
    setFeatureVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(FeatureFeedHeader);
