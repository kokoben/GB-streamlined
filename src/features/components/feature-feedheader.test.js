import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FeatureFeedHeader } from './feature-feedheader';

describe('feature-feedheader tests', () => {
  it('renders correctly', () => {
    const mockFetchFeatureSearchVideos = jest.fn();
    const mockSetFeatureVideo = jest.fn();
    const mockSetVideo = jest.fn();

    const wrapper = shallow(<FeatureFeedHeader
      currentFeatureSearchResults={[]}
      currentResults={[]}
      fetchFeatureSearchVideos={mockFetchFeatureSearchVideos}
      setFeatureVideo={mockSetFeatureVideo}
      setVideo={mockSetVideo}
    />);

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});

