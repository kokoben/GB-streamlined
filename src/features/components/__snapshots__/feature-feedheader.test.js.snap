// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature-feedheader tests renders correctly 1`] = `
<FeedHeader
  category="Feature"
  currentResults={Array []}
  fetchVideos={[Function]}
  placeholder="Search for Feature videos..."
  results={Array []}
  setVideo={[Function]}
/>
`;
